setup-pyenv.sh

```diff
@@ -10,19 +10,81 @@
 #     Directory in which to install pyenv [default: ~/.pyenv]
 # - PYENV_RELEASE
 #     Release tag of pyenv to download [default: clone from master]
-# - PYTHON_BUILD_CACHE_PATH:
-#     Directory in which to cache PyPy builds [default: ~/.pyenv_cache]
+# - PYENV_CACHE_PATH
+#     Directory where full Python builds are cached (i.e., for Travis)
+
+PYENV_ROOT="${PYENV_ROOT:-$HOME/.pyenv}"
+PYENV_CACHE_PATH="${PYENV_CACHE_PATH:-$HOME/.pyenv_cache}"
+version_cache_path="$PYENV_CACHE_PATH/$PYENV_VERSION"
+version_pyenv_path="$PYENV_ROOT/versions/$PYENV_VERSION"
+
+# Functions
+#
+# verify_python -- attempts to call the Python command or binary
+# supplied in the first argument with the --version flag. If
+# PYENV_VERSION_STRING is set, then it validates the returned version string
+# as well (using fgrep). Returns whatever status code the command returns.
+verify_python() {
+  local python_bin="$1"
```

Comment:
Personally I like to do:
```
local python_bin="$1"; shift
```
so that if the parameter isn't provided, then the function fails (at least with `set -e`).